lib/ansible/plugins/inventory/ini.py

```diff
@@ -180,7 +180,17 @@ def _parse(self, path, lines):
                     # but [groupname:vars] is allowed only if the # group is declared elsewhere.
                     # We add the group anyway, but make a note in pending_declarations to check at the end.
                     if state == 'vars':
-                        pending_declarations[groupname] = dict(line=self.lineno, state=state, name=groupname)
+
+                        # It's possible that a group is previously pending due to being
+                        # defined as a child group, in that case we simply pass so that
+                        # the logic below to process pending declarations will take the
+                        # appropriate action for a pending child group instead of
+                        # incorrectly handling it as a var state pending declaration
+                        if groupname in pending_declarations:
+                            if pending_declarations[groupname]['state'] == 'children':
```

Comment:
not sure this is correct, in theory there sould only be one reason for being in pending_declarations so we should not need the extra check for 'children' ... but now I'm thinking of dupicate vars entries ....

Reply:
```
[web:children]
appnodes
proxies

[web:vars]
deployment_type=prod

[appnodes:vars]
foo_var=true

[appnodes]
appnodes[1:3].example.com

[proxies:vars]
bar_var=true

[proxies]
proxies[1:3].example.com
```

In this example, the firs state processed for the declaration of either child group is "children", however it never gets processed because before the child group is actually defined, it is defined by a `var` state which results in the entry being deleted from the `pending_declarations` which is what lead me to that as the fix and it does work.